build.gradle

```diff
@@ -109,16 +109,19 @@ subprojects {
     }
 
     // Added for enforcing Java 1.6 version compatibility.
-    // Note that this will not result in compilation failure unless JDK 1.6 or less is used.
-    // GraalVM supports from JDK 1.8; therefore, it doesn't have an effect except IDE inspector
-    // may show incompatible APIs for higher version than JDK 1.6.
     tasks.withType(JavaCompile) {
-        def javaVersion = JavaVersion.VERSION_1_6
-        sourceCompatibility = javaVersion
-        targetCompatibility = javaVersion
-        options.bootstrapClasspath = files("$System.env.JAVA_HOME/jre/lib/rt.jar")
+            def jdkpath = project.property('jdkPath')
+            def folder = new File(jdkpath)
+            if(!folder.exists()) {
+                throw new GradleException("JDK 1.6 path does not exist ! Please set the correct 'jdkPath' in gradle.properties file")
+            }
+
```

Comment:
I think, we should execute `java --version` using this JDK(the one with jdk6Path) internally, extract version and check if it is  1.6 or not.  If not, throw exception with string - "jdk6Path project property is set with JDK 1.x path. Please set it with JDK 1.6 path."